File: 242-Valid-Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        counter = Counter(s)

        for c in t:
            if c not in counter or counter[c] == 0:
                return False
            counter[c] -= 1

        return True

File: 1_arrays-hashing/242-Valid-Anagram.ts
function isAnagram(s: string, t: string): boolean {
  const f = {}

  for (const c of s) {
    if (f[c]) f[c]++
    else f[c] = 1
  }

  for (const c of t) {
    if (!f[c]) return false
    else f[c]--
  }

  for (const key in f) if (f[key]) return false

  return true
};